knative-operator/pkg/controller/knativekafka/knativekafka_controller_test.go

```diff
@@ -711,3 +716,101 @@ func TestCheckHAComponent(t *testing.T) {
 		})
 	}
 }
+
+func TestMonitoringResources(t *testing.T) {
+
+	kk := &v1alpha1.KnativeKafka{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      "knative-kafka",
+			Namespace: "knative-eventing",
+		},
+		Spec: v1alpha1.KnativeKafkaSpec{
+			Broker:  v1alpha1.Broker{Enabled: true},
+			Source:  v1alpha1.Source{Enabled: false},
+			Sink:    v1alpha1.Sink{Enabled: true},
+			Channel: v1alpha1.Channel{Enabled: false},
+		},
+	}
+
+	t.Setenv("KAFKACHANNEL_MANIFEST_PATH", "../../../deploy/resources/knativekafka/channel")
+	t.Setenv("KAFKASOURCE_MANIFEST_PATH", "../../../deploy/resources/knativekafka/source")
+	t.Setenv("KAFKACONTROLLER_MANIFEST_PATH", "../../../deploy/resources/knativekafka/controller")
+	t.Setenv("KAFKABROKER_MANIFEST_PATH", "../../../deploy/resources/knativekafka/broker")
+	t.Setenv("KAFKASINK_MANIFEST_PATH", "../../../deploy/resources/knativekafka/sink")
+
+	r, err := newReconciler(&MockManager{})
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	manifests, err := r.buildManifest(kk, manifestBuildEnabledOnly)
+	if err != nil {
+		t.Fatal(err)
+	}
+
+	crGvk := schema.GroupVersionKind{
+		Group:   rbacv1.SchemeGroupVersion.Group,
+		Version: rbacv1.SchemeGroupVersion.Version,
+		Kind:    "ClusterRoleBinding",
+	}
+	svGvk := schema.GroupVersionKind{
+		Group:   monitoringv1.SchemeGroupVersion.Group,
+		Version: monitoringv1.SchemeGroupVersion.Version,
+		Kind:    "ServiceMonitor",
+	}
+	svcGvk := schema.GroupVersionKind{
+		Group:   corev1.SchemeGroupVersion.Group,
+		Version: corev1.SchemeGroupVersion.Version,
+		Kind:    "Service",
+	}
+
+	expected := map[schema.GroupVersionKind]sets.String{
+		crGvk: sets.NewString(
+			"rbac-proxy-reviews-prom-rb-kafka-controller",
+			"rbac-proxy-reviews-prom-rb-kafka-webhook-eventing",
+			"rbac-proxy-reviews-prom-rb-knative-kafka-broker-data-plane",
+			"rbac-proxy-reviews-prom-rb-knative-kafka-sink-data-plane",
+		),
+		svGvk: sets.NewString(
+			"kafka-controller-sm",
```

Comment:
Can we keep components in a list so we dont need to repeat them here in tests. That means we can simply pass the list when rbac proxy transformation is applied.